### PR TITLE
[VELOCITY-1190] Wait on app health during install_package

### PR DESCRIPTION
**Pull request by mgummelt**
cc @sschneid for review

### CODE REVIEW

**Comment by kensipe** [comment]
@mgummelt @sschneid I like the addition of `is_app_healthy` however after some thought  I'm now of the opinion that this wait is inappropriate.   The idea of "waiting" is covered in our spinners and perhaps we need a wait for health.

The `install_package_and_wait` function is challenging because it is 1 function which conflates 2 ideas and we are left with the question of wait for what.   The most logic event to wait for is deployment... the stages after deployment are also potentially interesting.. however they are use case driven.   The first logical step to wait for is deployed... after that a developer may be interested in health or a service endpoint.

With that justification, I believe the original code was incorrect (it waited to see that a FW was added to mesos), however I don't think this implementation is correct either.   It should wait for deployment to finish, not for health... either that or we should pass in the lambda of the wait function :)

I will review as is, with an expectation that we will have a conversation.

File: shakedown/dcos/package.py [hunk 1]
```diff
@@ -103,17 +103,10 @@ def install_package(
 
     # Optionally wait for the service to register as a framework
```
Comment (kensipe) [inline shakedown/dcos/package.py]: this comment is not consistent with the code that follows

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by mgummelt** [comment]
I do agree that health isn't the right check, but I don't think deployment completion is either.  The client probably cares about whether the package is ready to use, which is defined by a readiness check.  How about I change this to a readiness check (`is_app_ready`)?  @kensipe

**Comment by kensipe** [comment]
@mgummelt  In marathon, the app is "in deployment" until the readiness check passes.  Waiting from deployment to complete does provide the benefit you are looking for.

The challenge still remains... just because a service is fully deployed (and the readiness check passes) does NOT mean the service is accessible.   There can still be a small amount of time between a service readiness and DCOS configuring it's service endpoint.    If invoking behavior against the service is what you are after it is best to wait for the service endpoint.

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

**Comment by mgummelt** [comment]
@kensipe Ah yea, I didn't know deployments blocked on readiness.  Checking deployment should be fine, then. 

And yes, the adminrouter endpoint is async (and we block on that separately to determine if Spark is accessible), but services are often only accessed internally (e.g. HDFS), in which case it's sufficient to just block on readiness.

**Comment by mgummelt** [comment]
@kensipe Ready for review

**Comment by sschneid** [comment]
@mesos-ci retest this please

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

**Comment by kensipe** [comment]
code looks great!  tests are failing... I'll merge when the test passes.  Thanks for the collaboration on this... I definitely feel the end result is better as a result.

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by kensipe** [comment]
and sorry for the delay.. I didn't catch your comments prior to leaving HH

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by kensipe** [comment]
I ran locally with no failures

@mesosphere-ci retest this please

**Comment by kensipe** [comment]
updated some tests... these tests were flaky... it isn't clear that mgummelt changes had much to do with these failures... the tests needed fixing regardless.  there is more to fix but these should resolve the current challenge.

**Pull request merged**